presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -442,6 +440,21 @@ public synchronized void updatePartitionStatistics(HiveIdentity identity, Table
                 .collect(toImmutableList());
     }
 
+    private synchronized List<String> listAllTables(String databaseName)
+    {
+        requireNonNull(databaseName, "databaseName is null");
+
+        Optional<Database> database = getDatabase(databaseName);
+        if (database.isEmpty()) {
+            return ImmutableList.of();
+        }
+
+        Path databaseMetadataDirectory = getDatabaseMetadataDirectory(databaseName);
```

Comment:
This variable doesn't seem necessary

Reply:
i split out extract method as a separate commti